FIX adjust inliner criterion in RANSACRegressor (#19499)
scikit-learn/scikit-learn: doc/whats_new/v1.0.rst

  :pr:`19426` by :user:`Alexandre Gramfort <agramfort>` and
  :user:`Maria Telenczuk <maikia>`.

> - |Fix| :class:`RANSACRegressor` Points with a residual equal to 
>   ``residual_threshold`` member are considered as an inlier.
>   Motivation for this change: We want a perfect dataset to be considered as 
>   inliers, specifically with default parameters or ``residual_threshold`` 
>   equal to zero.
>   Previously the code launched an exceptions with perfect model.
>   Such a comportement also happen with MAD and half of the 
>   points perfect.
>   With this definition, ``residual_threshold`` set to zero no longer set all 
>   points to outliers. Now you can do that by setting ``residual_threshold`` to 
>   nan.
>   :pr:`19499` by :user:`Gregory Strubel <gregorystrubel>`.

Reviewer: I think this can be made shorter: readers interested by the details of those changes can refer to this PR.

```suggestion
- |Fix| Points with a residual equal to  ``residual_threshold`` are now considered
  as an inliers for :class:`linear_model.RANSACRegressor`. This allows fitting
  model perfectly on some datasets when ``residual_threshold=0``.
  :pr:`19499` by :user:`Gregory Strubel <gregorystrubel>`.
```

  :pr:`19426` by :user:`Alexandre Gramfort <agramfort>` and
  :user:`Maria Telenczuk <maikia>`.

> - |Fix| Points with a residual equal to  ``residual_threshold`` are now considered

Reviewer 2: ```suggestion
- |Fix| Points with residuals equal to  ``residual_threshold`` are now considered
```

>   as an inliers for :class:`linear_model.RANSACRegressor`. This allows fitting

Reviewer 2: ```suggestion
  as inliers for :class:`linear_model.RANSACRegressor`. This allows fitting
```

>   model perfectly on some datasets when ``residual_threshold=0``.

Reviewer 2: ```suggestion
  a model perfectly on some datasets when `residual_threshold=0`.
```
